File: TransCoder_tokenized_test_set_functions/python/FIND_NUMBER_PAIRS_ARRAY_XOR_0.py
def calculate ( a ) :
    a . sort ( )
    count = 1
    answer = 0
    for i in range ( 1, len ( a ) ) :
        if a [ i ] == a [ i - 1 ] :
            count += 1
        else :
            answer = answer + count * ( count - 1 ) //2
            count=1
    answer=answer+count*(count-1)//2
    returnanswer

File: results/py/FIND_NUMBER_PAIRS_ARRAY_XOR_0.cpp
int f_filled ( string a ) {
  sort ( a . begin ( ) , a . end ( ) ) ;
  int count = 1 ;
  int answer = 0 ;
  for ( int i = 1 ;
  i < a . length ( ) ;
  i ++ ) {
    if ( a [ i ] == a [ i - 1 ] ) count ++ ;
    else {
      answer = answer + count * ( count - 1 ) / 2 ;
      count = 1 ;
    }
  }
  answer = answer + count * ( count - 1 ) / 2 ;
  return answer ;
}
